src/packages/sdk/src/getPackedPackage.ts

```diff
@@ -44,12 +47,24 @@ export async function getPackedPackage(
   const tmpDir = tempy.directory() // thanks Sindre
   const archivePath = path.join(tmpDir, `package.tgz`)
 
+  // Check if yarn is available.
+  const isYarn = await hasYarn(packageDir);
+
+  const packCMD = isYarn ? `yarn pack -f ${archivePath}` : `npm pack ${packageDir}`;
+
   // pack into a .tgz in a tmp dir
-  await execa.command(`yarn pack -f ${archivePath}`, {
+  await execa.command(packCMD, {
     shell: true,
-    cwd: packageDir,
+    cwd: isYarn ? packageDir : tmpDir, // for npm pack it outputs a file to the cwd
   })
 
+  if (!isYarn) { // since npm pack does not have option to specify a filename we change it here
+    // find single tgz in temp folder
+    const filename = (await readdir(tmpDir))[0]
```

Comment:
From just reading this, can't it be, that `npm pack` will create a `.tgz` in a folder that already has files? If there would be a file that would be read first, we would have the wrong file here

Reply:
if we assume that tmp directory is the same - yes,
but as it is stated in docs. of tempy library, that is being used to create a tmp dir.:
```
tempy.directory(options?)
Get a temporary directory path. **The directory is created for you.**
```
it will create a unique directory every time, leading to the empty directory, so practically this cannot happen.

Reply:
Ok, but isn't `npm pack` generating to `packageDir` here, which is not the `tmp` dir, but the actual directory of the package?

Reply:
no it is not. `npm pack` is generating tgz to the `cwd` of the process and on line 58
```
cwd: isYarn ? packageDir : tmpDir, // for npm pack it outputs a file to the cwd
```
i'm setting `cwd` to be `tmpDir`.
Meaning if u do something like: `cd /tmp/ && npm pack /abs/path/to/package/` it will output tgz into the `tmp` directory.

Reply:
Ok now I get it. Thanks for clarifying!